### DISCUSSION TITLE
Middleware is blocking redirect after login

### DISCUSSION THREAD

**Discussion by bilel-lm** [nextauthjs/next-auth#7492] [Help]
### Environment

"next": "13.4.0",
"next-auth": "^4.22.1",
"react": "18.2.0"

OS: windows/macOS

Node: 18.15.0

### Reproduction URL

no public repository

### Describe the issue

Hey I'm using next-auth and next-intl (a translation package)
I had to use the advanced method of the middleware to be able to block the application to non connected users.
However when I log in successfully I am not always redirected to the home page. (redirecting with router.push)
The token is null in the middleware and it considers me as not authenticated, but I can see my username in the nav bar at the top.

### How to reproduce

```tsx
import { withAuth } from "next-auth/middleware";
import createMiddleware from "next-intl/middleware";
import { NextMiddlewareResult } from "next/dist/server/web/types";

import {
  NextFetchEvent,
  NextMiddleware,
  NextRequest,
  NextResponse,
} from "next/server";

const locales = ["en", "de"];
const publicPages = ["/login"];

const intlMiddleware = createMiddleware({
  locales: locales,
  defaultLocale: "en",
});

type NextMiddlewareWithRequest = (request: NextRequest) => NextMiddlewareResult;
const authMiddleware = (
  req: NextRequest,
  event: NextFetchEvent,
  otherMiddleware?: NextMiddlewareWithRequest
) =>
  (
    withAuth(
      (r: NextRequest) =>
        otherMiddleware ? otherMiddleware(r) : NextResponse.next(),
      {
        callbacks: {
          authorized: ({ token }) => {
            console.log("token", token);
            return Boolean(token);
          },
        },
        pages: {
          signIn: "/login",
        },
      }
    ) as NextMiddleware
  )(req, event);

export default function middleware(req: NextRequest, event: NextFetchEvent) {
  const publicPathnameRegex = RegExp(
    `^(/(${locales.join("|")}))?(${publicPages.join("|")})/?$`,
    "i"
  );
  const isPublicPage = publicPathnameRegex.test(req.nextUrl.pathname);

  if (isPublicPage) {
    return intlMiddleware(req);
  } else {
    return authMiddleware(req, event, intlMiddleware);
  }
}

export const config = {
  matcher: ["/((?!api|_next|.*\\..*).*)"],
};
```

### Expected behavior

User logged in but not always redirected after successfull login

**Comment by SFPink** [comment]
Experiencing this also. 

On login no redirect happens, url will update to include the callback query with the intended redirect url as the value, navigation bar updates to show someone has in fact logged in. If I refresh everything works are normal for that user.

When calling signIn I am not using the built in callback functionality, redirect is disabled. I am doing a router.push to redirect.

I should also note this only happens when my project is built for production, in development it works as intended.

**Comment by SFPink** [comment]
I ended up going with a temporary solution and used `location.href = {{url}}` instead of `router.push()`, because on refresh everything worked as intended. It's not elegant but it works.

**Reply by bilel-lm** [reply]
Thanks for the solution, unfortunately it still doesn't work for me.

**Comment by edikamy** [comment]
Hi
I have a problem like this
I build a sandbox to visualize the problem

https://codesandbox.io/p/sandbox/next-intl-bug-template-app-forked-rk5jpr?file=%2Fsrc%2Futils%2FNextAuth%2FdefaultOptions.ts%3A34%2C8

in my project you can see token from the "authorized" callback always is null but in "defaultOptions", if you fill "admin" in username and password you should get access_token

**Comment by bilel-lm** [comment]
Hello, can someone please help me? I'm still stuck on this issue 😭
